consensus/polybft/block_builder.go

```diff
@@ -221,7 +213,6 @@ func (b *BlockBuilder) writeTxPoolTransaction(tx *types.Transaction) (bool, erro
 
 	// remove tx from the pool and add it to the list of all block transactions
 	b.params.TxPool.Pop(tx)
-	b.txns = append(b.txns, tx)
```

Comment:
I think we gonna need this because we lose some functionality if we simple call `err := b.WriteTx(tx)` (mentioned in previous comment)

Reply:
If `WriteTx` is successful (there is no error), it appends the transaction to the block builder. No need to have it in this function.